cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -103,26 +138,48 @@ public InMemoryConfigurator(CConfiguration cConf, Id.Namespace appNamespace, Id.
    */
   @Override
   public ListenableFuture<ConfigResponse> config() {
-    SettableFuture<ConfigResponse> result = SettableFuture.create();
 
+    // Create the classloader
+    ClassLoader tempClassLoader = artifactClassLoader;
+    if (tempClassLoader == null) {
+      EntityImpersonator classLoaderImpersonator = new EntityImpersonator(artifactId.toEntityId(), impersonator);
+      try {
+        tempClassLoader = artifactRepository.createArtifactClassLoader(artifactLocation, classLoaderImpersonator);
+      } catch (Exception e) {
+        return Futures.immediateFailedFuture(e);
+      }
+    }
+
+    CloseableClassLoader classLoader = (CloseableClassLoader) tempClassLoader;
```

Comment:
Is this casting always safe? Seems like instead of casting, you can just do a instanceof check in the `finally` block.

```
try {
  ///....
} finally {
  if (artifactClassLoader == null && tempClassLoader instanceof Closeable) {
    Closeables.closeQuietly((Closeable) tempClassLoader);
  }
}
```

Reply:
Done